api/lib/domain/events/handle-clea-certification-scoring.js

```diff
@@ -1,47 +1,62 @@
 const { checkEventTypes } = require('./check-event-types');
 const CertificationScoringCompleted = require('./CertificationScoringCompleted');
+const CertificationRescoringCompleted = require('./CertificationRescoringCompleted');
+const { featureToggles } = require('../../config');
+const { CLEA } = require('../models/ComplementaryCertification');
 
-const eventTypes = [CertificationScoringCompleted];
+const eventTypes = [CertificationScoringCompleted, CertificationRescoringCompleted];
 
 async function handleCleaCertificationScoring({
   event,
   partnerCertificationScoringRepository,
   badgeRepository,
   certificationCourseRepository,
+  cleaCertificationResultRepository,
   certificationCenterRepository,
   knowledgeElementRepository,
   targetProfileRepository,
   badgeCriteriaService,
+  complementaryCertificationCourseRepository,
 }) {
   checkEventTypes(event, eventTypes);
   const { certificationCourseId, userId, reproducibilityRate } = event;
 
+  if (featureToggles.isComplementaryCertificationSubscriptionEnabled) {
+    return _handleToggleEnabled(
+      complementaryCertificationCourseRepository,
+      certificationCourseId,
+      partnerCertificationScoringRepository,
+      userId,
+      reproducibilityRate
+    );
+  }
+
   const certificationCenter = await certificationCenterRepository.getByCertificationCourseId(certificationCourseId);
   if (!certificationCenter.isHabilitatedClea) {
     return;
   }
 
-  const cleaCertificationScoring = await partnerCertificationScoringRepository.buildCleaCertificationScoring({
-    certificationCourseId,
-    userId,
-    reproducibilityRate,
-  });
-
-  if (cleaCertificationScoring.hasAcquiredBadge) {
-    await _verifyBadgeValidity(
-      certificationCourseRepository,
+  if (event instanceof CertificationRescoringCompleted) {
+    return _handleRescoringToggleDisabled(
```

Comment:
```suggestion
    return _handleRescoringWithToggleDisabled(
```
pour être cohérent avec `_handleScoringWithToggleDisabled`